src/Utils/DelayedUnmount.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useEffect, useLayoutEffect, useState } from 'react';
+
+interface Props {
+  mount: boolean;
+  delay?: number;
+}
+
+export const DelayedUnmount: React.FC<Props> = ({
+  children,
+  mount,
+  delay = 300,
```

Comment:
This might be redundant since we have hard-coded it `290`

Reply:
Well I was trying to keep this component general so that it can be reused in other places too.